docs/content/en/docs/concepts/architecture/components/scheduler/_index.md

```diff
@@ -6,11 +6,50 @@ weight: 80
 cascade:
 ---
 
-**Keptn's Lifecycle Scheduler** replaces the
-[Kubernetes scheduler](https://kubernetes.io/docs/concepts/scheduling-eviction/kube-scheduler/)
-to allow users to schedule events and tasks to occur
-at specific times during the application lifecycle.
-The Lifecycle Scheduler can trigger events such as
-deployment, testing, and remediation at specific times or intervals.
-The Keptn Scheduler guarantees that Pods are initiated only after
-the Pre-Deployment checks are completed.
+The **Keptn Scheduler**, an integral component of the Keptn Lifecycle Toolkit which orchestrates
+the deployment process.
+The **Keptn Scheduler** works by registering itself as a Permit plugin with Kubernetes that
+is responsible for ensuring that Pods are scheduled to a node until and unless the
+pre-deployment checks have finished successfully.
+This is important because it helps to prevent Pods from being scheduled to nodes that are
+not yet ready for them, which can lead to errors and downtime
+
+## How does the Keptn Scheduler works
+
+The Keptn Scheduler first checks the annotations on the Pod to see if it is annotated with
+[Keptn specific annotations](https://main.lifecycle.keptn.sh/docs/implementing/integrate/#basic-annotations).
+If the Pod is annotated with Keptn specific annotations, the Keptn Scheduler retrieves
+the WorkloadInstance CRD that is associated with the Pod.
+The **WorkloadInstance CRD** contains information about the `pre-deployment` checks that
+need to be performed before the Pod can be scheduled.
+
+The Keptn Scheduler then checks the status of the WorkloadInstance CRD to see if the
+`pre-deployment` checks have finished successfully.
+If the pre-deployment checks have finished successfully, the **Keptn Scheduler** allows
+the Pod to be scheduled to a node.
+If the `pre-deployment` checks have not yet finished, the Keptn Scheduler tells Kubernetes to check again later.
+
+The Keptn Scheduler processes the following information from the WorkloadInstance CRD:
+
+- The name of the pre-deployment checks that need to be performed.
+- The status of the pre-deployment checks.
+- The deadline for the pre-deployment checks to be completed.
+
+The Keptn Scheduler uses this information to determine whether or not the Pod can be scheduled.
+If all of the pre-deployment checks have finished successfully and the deadline has not been reached,
+the Keptn Scheduler allows the Pod to be scheduled.
+If any of the `pre-deployment` checks have not finished successfully or the deadline has
+been reached, the Keptn Scheduler tells Kubernetes to check again later.
+
+The Keptn Scheduler also has the ability to blacklist nodes that have failed pre-deployment checks.
```

Comment:
I do not agree with this sentence, this may be a k8s default behavior. (not sure) but we do not black list nodes. I would remove this part